lib/assert.js

```diff
@@ -103,8 +104,21 @@ x.throws = function (fn, err, msg) {
 			}
 		}, err, msg);
 
+		if (!isError(result)) {
+			throw result;
+		}
```

Comment:
This is necessary in case a developer writes this test

```js
test(t => {
    t.throws(() => {
        throw 'foo';
    });
});
```

`assert.throws` will validate that the function indeed throws an error (a string) and thus does not jump to line 113.